File: CommonBusiness/src/main/java/com/company/commonbusiness/base/fragment/BaseMvpFragment.java
package com.company.commonbusiness.base.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import com.company.commonbusiness.base.mvp.BasePresenter;
import com.company.commonbusiness.base.mvp.IMvpView;

/**
 * @author 李昭鸿
 * @desc: MVP模式的Fragment基类
 * @date Created on 2017/9/21 18:39
 */

public abstract class BaseMvpFragment<P extends BasePresenter> extends BaseFragment implements IMvpView {

	protected P presenter;

	@Override
	public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		setPresenter();
		if (presenter != null) {
			presenter.attachView(this);
		}
	}

	@Override
	public void onDestroyView() {
		super.onDestroyView();
		if (presenter != null) {
			presenter.detachView();
		}
	}
}

File: CommonBusiness/src/main/java/com/company/commonbusiness/base/fragment/BaseMvpFragment.kt
package com.company.commonbusiness.base.fragment

import com.company.commonbusiness.base.mvp.IMvpPresenter

/**
 * @author 李昭鸿
 * @desc: MVP模式的Fragment基类
 * @date Created on 2017/9/21 18:39
 * @github: https://github.com/lisuperhong
 */

abstract class BaseMvpFragment<P : IMvpPresenter> : BaseFragment() {

    protected var presenter: P? = null

    override fun onDestroyView() {
        super.onDestroyView()
        if (presenter != null) {
            presenter!!.detachView() //释放资源
        }
        presenter = null
    }

    /**
     * 绑定Presenter
     */
    protected abstract fun setPresenter()
}